src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs

```diff
@@ -127,6 +129,51 @@ public static CommandLineBuilder CancelOnProcessTermination(this CommandLineBuil
             return builder;
         }
 
+        public static CommandLineBuilder AbandonOnRepeatCancellation(
+            this CommandLineBuilder builder,
+            int repeatThreshold = 1)
+        {
+            builder.AddMiddleware(async (context, next) =>
+            {
+                var initialCancelToken = context.GetCancellationToken();
```

Comment:
We can't call `GetCancellationToken` here. When the `CancellationToken` gets requested (which is assumed to be done by the user), the context assumes that `CancellationToken` will be used to implement `Ctrl+C` handling, and the app no longer terminates by default.

So this has the unintended side-effect that command handlers which don't use a `CancellationToken` no longer terminate on first `Ctrl+C`.